kafka/stats_event_test.go

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+)
+
+// handleStatsEvent checks for stats event and signals on statsReceived
+func handleStatsEvent(t *testing.T, eventCh chan Event, statsReceived chan bool) {
+	for ev := range eventCh {
+		switch e := ev.(type) {
+		case *Stats:
+			t.Logf("Stats: %v", e)
+
+			// test if the stats string can be decoded into JSON
+			var raw map[string]interface{}
+			err := json.Unmarshal([]byte(e.String()), &raw) // convert string to json
+			if err != nil {
+				t.Fatalf("json unmarshall error: %s", err)
+			}
+			t.Logf("Stats['name']: %s", raw["name"])
+			close(statsReceived)
```

Comment:
@edenhill If the `eventCh` has more than one `Stats` messages, then `statsReceived ` would be closed twice, which should panic the program. I wonder how this actually can work ? In my understanding, golang channel cannot be closed twice.